Add a few small ruins

babydoll opened the pull request:
:cl:
maptweak: Added a few small ruin sites to make exoplanets a little more detailed.
/:cl:

3 oases
1 fountain thing

babydoll committed:
Add a few small ruin sites

CakeQ commented on code/modules/overmap/exoplanets/desert.dm:
```diff
@@ -2,6 +2,7 @@
 	name = "desert exoplanet"
 	desc = "An arid exoplanet with sparse biological resources but rich mineral deposits underground."
 	color = "#d6cca4"
+	possible_features = list(/datum/map_template/ruin/exoplanet/monolith, /datum/map_template/ruin/exoplanet/oasis, /datum/map_template/ruin/exoplanet/oasis/oasis2, /datum/map_template/ruin/exoplanet/oasis/oasis3)
```
God this is going to get messy really quick. We should have some form of per-feature planet flag and have a system run through all features to create lists on runtime

babydoll replied:
I agree but beyond the scope of this PR which is to add three 12x7 maps

MistakeNot4892 replied:
Maybe break it up into multiple lines for the sake of readability, agree that redoing the system is out of scope though.

babydoll commented:
This is WIP waiting for #21849 so there aren't any conflicts, by the way. The maps themselves are done

Techhead0 commented:
If the oases don't have anything desert-specific in them, you can probably stick them on other planet types as ordinary ponds.

babydoll commented:
That's a good point, the sand doesn't have to be there

babydoll commented:
Removed the sand from the one map, added a little fountain site too. I made the map too big for the fountain so once I go back to add exceptions everywhere I'll trim the fat from fountain_ruin.dmm

afterthought2 commented on maps/random_ruins/exoplanet_ruins/oasis/oasis.dm:
```diff
@@ -0,0 +1,18 @@
+	cost = 0.5
+
+/datum/map_template/ruin/exoplanet/oasis/oasis2
+	id = "oasis2"
```
One thing that the ruin loader supports is having several templates share the same id. The effect is that at most one of those templates will be chosen (globally). Not saying to do it here, necessarily, just fyi.

Actually, while it was clearly part of the original design, the way it's implemented doesn't work. Could do a round of tweaks later.

babydoll committed:
Add exceptions and set planets to use new ruins

afterthought2 commented on maps/random_ruins/exoplanet_ruins/fountain/fountain_ruin.dmm:
```diff
@@ -0,0 +1,82 @@
+//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE
+"a" = (
+/turf/template_noop,
+/area/map_template/fountain)
```
Unless I'm missing the define, you may need to/meant to remove this from the dmm in favor of area/template_noop. Same for the the other dmms.

babydoll replied:
oops, didn't reopen the maps so forgot about this

afterthought2 commented:
You'll want to rerun Travis on this, as due to the bug I created earlier it was doing nothing on exoplanets for a little while.
